content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -15,6 +15,48 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
 
 <!-- more -->
 
+## Deferred Rendering
+
+<div class="release-feature-authors">authors: @DGriffin91</div>
+
+The two most popular "rendering styles" are:
+
+* **Forward Rendering**: do all material/lighting calculations in a single render pass
```

Comment:
Forward rendering vs deferred:
* Forward can render transparencies, deferred cannot, also only forward can do MSAA (forward wins)
* Deferred saves on lighting costs - you only shade what's visible, where with forward you may be shading pixels that will be occluded later on (deferred wins)
* Deferred involves a lot of texture bandwidth and multiple render targets, which is prohibitive on mobile (forward wins)

In general I would say forward + small prepass (depth, +normals if using SSAO) is the way to go, but deferred opens up specific niches such as real-time GI (screenspace or otherwise) that needs a full G-buffer, and saves on costs when your material lighting shader is significantly expensive or would destroy register pressure in a forward shader.

"Number of lights" isn't that big a difference, you can do clustering in either one.

Visibility buffers are another possibility to further reduce overhead, as it _fully_ decouples "is this pixel visible" from _any_ sort of lighting, including material evaluation such as writing out a gbuffer pixel. Adding occlusion culling further cuts down on a lot of over-shading issues. I intend to experiment with this in my meshlet work for 0.13.

For now, forward is the better default, and users should experiment with deferred only for specific situations where it makes sense.

Reply:
> Forward can render transparencies, deferred cannot

I tried to avoid absolute statements here. I'm pretty sure order-independent transparency works for deferred?

Reply:
True, I don't know anything about OIT. Best to err on the side of caution.